### PR TITLE
[Modules] Update - Microsoft.VirtualMachineImages/images to include stagingResourceGroup with new API Version

### PR DESCRIPTION
**Pull request by shawntmeyer**
# Description

Updated VirtualMachineImages/Images template (Azure VM Image Builder template) to the latest API to include the new capability for a named staging resource group. By naming a staging resource group we can avoid issues with Azure Policies that prevent storage accounts and key vaults without Private Endpoints by creating a permanent exclusion for the named resource group.

# Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

File: modules/Microsoft.VirtualMachineImages/imageTemplates/deploy.bicep [hunk 1]
```diff
@@ -42,6 +42,9 @@ param sigImageDefinitionId string = ''
 @description('Optional. List of the regions the image produced by this solution should be stored in the Shared Image Gallery. When left empty, the deployment\'s location will be taken as a default value.')
 param imageReplicationRegions array = []
 
+@description('Optional. The staging resource group id in the same subscription as the image template that will be used to build the image. If this field is empty, a resource group with a random name will be created. If the resource group specified in this field doesn\'t exist, it will be created with the same name. If the resource group specified exists, it must be empty and in the same region as the image template. The resource group created will be deleted during template deletion if this field is empty or the resource group specified doesn\'t exist, but if the resource group specified exists the resources created in the resource group will be deleted during template deletion and the resource group itself will remain.')
+param stagingResourceGroup string
```
Comment (AlexanderSehr) [inline modules/Microsoft.VirtualMachineImages/imageTemplates/deploy.bicep:46]: If optional - we should provide a default value ;)

Comment (AlexanderSehr) [reply modules/Microsoft.VirtualMachineImages/imageTemplates/deploy.bicep:46]:
```suggestion
param stagingResourceGroup string = ''
```

File: modules/Microsoft.VirtualMachineImages/imageTemplates/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -89,5 +89,6 @@ module testDeployment '../../deploy.bicep' = {
     unManagedImageName: '<<namePrefix>>-umi-${serviceShort}-001'
     userMsiResourceGroup: resourceGroupName
     vmSize: 'Standard_D2s_v3'
+    stagingResourceGroup: ''
```
Comment (AlexanderSehr) [inline modules/Microsoft.VirtualMachineImages/imageTemplates/.test/common/deploy.test.bicep:92]: I guess - just for fun - we could provide a value?
```suggestion
    stagingResourceGroup: '<<namePrefix>>${serviceShort}001-rg'
```

**Comment by shawntmeyer** [comment]
closing to resync my fork's main branch

**Pull request closed without merging**